matplotcheck/tests/test_base_data.py

```diff
@@ -68,159 +112,116 @@ def test_assert_xydata_floatingpoint_error(pt_scatter_plt, pd_df):
 
 
-def test_assert_xydata_xlabel_numeric_fails():
-    "Tests the xlabels flag on xydata correctly fails with numeric expected x-labels."
-    correct_data = {
-        "months": [1, 2, 3, 4, 5, 6, 7],
-        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
-    }
-    plot_data = {
-        "months": [1, 2, 3, 4, 5, 6, 99999],
-        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
-    }
-    correct_df = pd.DataFrame(correct_data)
-    plot_df = pd.DataFrame(plot_data)
+def test_assert_xydata_xlabel_numeric_fails(
+    pd_df_monthly_data_numeric, pt_monthly_data_numeric
+):
+    """Tests the xlabels flag on xydata correctly fails with numeric expected
+    x-labels."""
 
-    fig, ax = plt.subplots()
-    plot_df.plot("months", "data", kind="bar", ax=ax)
-    axis = plt.gca()
+    pd_df_expected_data = pd_df_monthly_data_numeric
+    pd_df_expected_data.loc[6, "months"] = 99999
 
-    pt = PlotTester(axis)
     with pytest.raises(AssertionError, match="Incorrect data values"):
-        pt.assert_xydata(correct_df, xcol="months", ycol="data", xlabels=True)
+        pt_monthly_data_numeric.assert_xydata(
+            pd_df_expected_data, xcol="months", ycol="data", xlabels=True
+        )
     plt.close()
 
 
-def test_assert_xydata_xlabel_numeric_fails_bad_y():
+def test_assert_xydata_xlabel_numeric_fails_bad_y(
+    pd_df_monthly_data_numeric, pt_monthly_data_numeric
+):
     """Tests that the xlabels flag on xydata correctly fails with wrong numeric
     y-data"""
-    "Tests the xlabels flag on xydata correctly fails with numeric expected x-labels."
-    correct_data = {
-        "months": [1, 2, 3, 4, 5, 6, 7],
-        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
-    }
-    plot_data = {
-        "months": [1, 2, 3, 4, 5, 6, 7],
-        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 9999],
-    }
-    correct_df = pd.DataFrame(correct_data)
-    plot_df = pd.DataFrame(plot_data)
 
-    fig, ax = plt.subplots()
-    plot_df.plot("months", "data", kind="bar", ax=ax)
-    axis = plt.gca()
+    pd_df_expected_data = pd_df_monthly_data_numeric
+    pd_df_expected_data.loc[6, "data"] = 99999
 
-    pt = PlotTester(axis)
     with pytest.raises(AssertionError, match="Incorrect data values"):
-        pt.assert_xydata(correct_df, xcol="months", ycol="data", xlabels=True)
+        pt_monthly_data_numeric.assert_xydata(
+            pd_df_expected_data, xcol="months", ycol="data", xlabels=True
+        )
     plt.close()
 
 
-def test_assert_xydata_xlabel_numeric_expected_string_actual():
+def test_assert_xydata_xlabel_numeric_expected_string_actual(
+    pd_df_monthly_data_numeric
+):
     """Tests the xlabels flag on xydata correctly fails with numeric expected
-    x-labels and non-numeric actial x-labels"""
-    correct_data = {
-        "months": [1, 2, 3, 4, 5, 6, 7],
-        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
-    }
+    x-labels and non-numeric actual x-labels"""
+
     plot_data = {
```

Comment:
looks like this could be a fixture too... just add the new values to the fixture that you call in this test that are needed to produce the desired plot_data. it might make sense to make this plot tests object a fixture too even tho it's only used once. it just makes things a bit cleaner.